src/common/schedulers/slurm_commands.py

```diff
@@ -70,6 +72,48 @@
 DEFAULT_UPDATE_COMMAND_TIMEOUT = 60
 
 
+class PartitionNodelistMapping:
+    """
+    Singleton class to represent the partition-nodelist mapping between PC-managed partitions and PC-managed nodes.
+
+    The information contained in this singleton is used to make sure ParallelCluster only handles partitions and nodes
+    it is supposed to manage (via Slurm commands).
+    """
+
+    partition_nodelist_mapping: Dict[str, str]
+    _instance = None
+
+    def __init__(self):
+        self.partition_nodelist_mapping = {}
+
+    def get_partition_nodelist_mapping(self) -> Dict[str, str]:
+        """Retrieve partition-nodelist mapping from JSON file."""
+        if not self.partition_nodelist_mapping:
+            partition_nodelist_json = os.path.join(
```

Comment:
I don't understand why Codecov complains about this: this should be covered by the new `test_get_partition_nodelist_mapping` defined below (see `test_slurm_commands.py`).

Reply:
We are not really mocking the reading of the file, we are only mocking the value of the global variable `SLURM_CONF_DIR` to be under the test data directory of pytest.

I believe Codecov is getting crazy here.